src/Carbon/CarbonPeriod.php

```diff
@@ -651,10 +651,12 @@ public function __construct(...$arguments)
                 $parsed = @CarbonInterval::make($argument)
             ) {
                 $this->setDateInterval($parsed);
-            } elseif ($this->startDate === null && $parsed = Carbon::make($argument)) {
-                $this->setStartDate($parsed);
-            } elseif ($this->endDate === null && $parsed = Carbon::make($argument)) {
-                $this->setEndDate($parsed);
+            } elseif ($parsed = Carbon::make($argument)) {
+                if ($this->startDate === null) {
+                    $this->setStartDate($parsed);
+                } elseif ($this->endDate === null) {
+                    $this->setEndDate($parsed);
+                }
```

Comment:
This makes the behavior change, that's why the unit test fails. I'm closing this but I would re-open it if you can revert this part and rebase.